xcodeproj/internal/product.bzl

```diff
@@ -98,14 +103,51 @@ exit ${PIPESTATUS[0]}
 
     return output
 
+def _extract_archive(*, actions, archive, bundle_name, bundle_extension):
+    output = actions.declare_directory(
+        bundle_name + bundle_extension,
+        sibling = archive,
+    )
+
+    args = actions.args()
+    args.add(archive)
+    args.add(output.path)
+
+    actions.run_shell(
+        inputs = [archive],
+        outputs = [output],
+        command = """\
+set -eu
+
+readonly archive="$1"
+readonly output="$2"
+
+if [[ "$archive" = *.ipa ]]; then
+    suffix=/Payload
+else
+    suffix=
+fi
+
+expanded_dir=$(mktemp -d)
+trap 'rm -rf "$expanded_dir"' EXIT
+
+unzip -q -DD "$archive" -d "$expanded_dir"
+mv "$expanded_dir$suffix/${output##*/}" "${output%/*}"
+""",
+        arguments = [args],
+        mnemonic = "XcodeProjExtractBundleArchive",
```

Comment:
This mnemonic should likely be added to here, right? Do we need to call it out in any docs for users?

https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/61dd744ea6806937b80330d81cac8fffdac1f9d6/shared.bazelrc#L53

Reply:
I was wondering if I should mirror one of the existing bundle ones. I did at first,but I thought it might make debugging stuff more difficult.

Maybe we just add the correct execution info here. Though I'm not sure we want to force it a certain way by default (though also, I think with modify_execution_info you can override the defaults?).

Reply:
I personally like it being separate. Just want it called out as a potential cache inflator that's not necessarily expected when onboarding/updating. And I'm not sure how `modify_execution_info` plays with defaults.